internal/api/handler.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	"github.com/labstack/echo/v4"
+	"github.com/pkg/errors"
+)
+
+// ItemReader is an interface for getting items
+type ItemReader interface {
+	GetAllItems(ctx context.Context) ([]models.Item, error)
+	GetStories(ctx context.Context) ([]models.Item, error)
+	GetJobs(ctx context.Context) ([]models.Item, error)
+}
+
+// Handler contains all the business logic for handling requests
+type Handler struct {
+	itemReader ItemReader
+	ctx        context.Context
+}
+
+// New returns a new api handler
+func New(itemReader ItemReader, ctx context.Context) *Handler {
+	return &Handler{
+		itemReader: itemReader,
+		ctx:        ctx,
+	}
+}
+
+// GetAllItems gets all the items in the item repository
+func (h *Handler) GetAllItems(c echo.Context) (err error) {
+	i, err := h.itemReader.GetAllItems(h.ctx)
```

Comment:
You can use echo's request context instead (`c.Request().Context()`). Doing this, in combination with the above comment, means that all of the functions called as part of a signle request share the same context.